### PR TITLE
Docs edits

### PR DESCRIPTION
**Pull request by danielballan**
Congrats on the release!

Reading through the API changes, I was impressed by the detailed instructions for rolling back any given style change (if possible). Very thorough work!

I found some small typos and a placeholder that probably was not meant to be included in the rc. If someone can point me to the relevant issue, I will add a commit properly replacing the placeholder (rather than just deleting it).

### CODE REVIEW

File: doc/users/dflt_style_changes.rst [hunk 1]
```diff
@@ -426,11 +426,11 @@ obscuring data too much.
     fig, (old, new) = plt.subplots(ncols=2, sharey=True)
     with plt.style.context('default'):
         new.boxplot(data, labels=['A', 'B', 'C', 'D'])
-        new.set_title('New boxplots')
```
Comment (jenshnielsen) [inline doc/users/dflt_style_changes.rst]: Looks like this (And below) is reversed?

Comment (danielballan) [inline doc/users/dflt_style_changes.rst]: Oops, yes. Fixed and force-pushed.

File: doc/users/dflt_style_changes.rst [hunk 2]
```diff
@@ -1130,10 +1130,3 @@ mplot3d
   - grid.color
   - grid.linewidth
   - grid.linestyle
-
-
-
-TEMPORARY NOTES TOM IS KEEPING IN THE SOURCE SO THEY DO NOT GET LOST
```
Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: this is here so I remember to do something about it!

**Comment by phobson** [comment]
@danielballan can you set the y-scales to `'log'` while your fixing this?

File: doc/users/dflt_style_changes.rst [hunk 3]
```diff
@@ -426,11 +426,11 @@ obscuring data too much.
...
         new.boxplot(data, labels=['A', 'B', 'C', 'D'])
-        new.set_title('New boxplots')
+        new.set_title('v2.0')
 
     with plt.style.context('classic'):
         old.boxplot(data, labels=['A', 'B', 'C', 'D'])
-        old.set_title('Old boxplots')
+        old.set_title('classic')
 
     new.set_ylim(bottom=0)
```
Comment (phobson) [inline doc/users/dflt_style_changes.rst]: If we log-scale the y-axes, this line will need to go.

Comment (danielballan) [inline doc/users/dflt_style_changes.rst]: Good idea, a log scale would make it easier to see the components here. Done.

**Comment by tacaswell** [comment]
backported to v2.x as f2c145e

File: doc/users/dflt_style_changes.rst (hunk 2 above)
Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: I'm just going to merge this as-is, an make a note on 'paper' to do something about this.

**Pull request merged**